tessera-core/src/main/java/com/quorum/tessera/node/PartyInfoParser.java

```diff
@@ -32,37 +34,49 @@ default PartyInfo from(final byte[] encoded) {
 
         final ByteBuffer byteBuffer = ByteBuffer.wrap(encoded);
 
-        final int urlLength = (int) byteBuffer.getLong();
+        final int urlLength = (int) toIntExact(byteBuffer.getLong());
+        checkLength(urlLength);
+
 
         final byte[] urlBytes = new byte[urlLength];
         byteBuffer.get(urlBytes);
         final String url = new String(urlBytes, UTF_8);
 
-        final int numberOfRecipients = (int) byteBuffer.getLong();
+        final int numberOfRecipients = (int) toIntExact(byteBuffer.getLong());
+        checkLength(urlLength);
```

Comment:
Should this be checkLength(numberOfRecipients); ?

Reply:
Well spotted Nicolae!